lib/LaTeXML/Core/Document.pm

```diff
@@ -998,7 +998,10 @@ sub floatToElement {
   my @candidates = getInsertionCandidates($$self{node});
   my $closeable  = 1;
   if (@candidates && $self->canContain($candidates[0], $qname)) {
-    return $candidates[0]; }    # Fine right here.
+    # Fine right here, just return
+    # -- special case, if current is a text node, float to its parent element and return
```

Comment:
Actually, this is exactly the problem. The comment could be made to reduce some of the mystery in the future. How about
```
# But don't resume at a text node, if it is current, and append more to it after other insertions.

Reply:
Funnily enough, the comment is hard to read for me... Is it:

1. "Don't resume at a text node, if it is current. Don't append more to it after other insertions."
2. "Don't resume at a text node, if it is current. Instead, append more to it after other insertions."

I think you mean 1. ? It's definitely an "Edge case" of sorts, can adapt the comment to that...

Reply:
you got it!